File: TILAB/src/main/java/com/TILab/TILAB/model/DetallePrestamo.java
package com.TILab.TILAB.model;

import jakarta.persistence.*;

@Entity
@Table(name = "detalles_prestamo")
public class DetallePrestamo {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "prestamo_id", nullable = false)
    private Prestamo prestamo;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "kit_id")
    private Kit kit;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "componente_id")
    private Componente componente;
    
    @Column(nullable = false)
    private Integer cantidad;
    
    @Column
    private String numeroSerie;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Prestamo getPrestamo() {
        return prestamo;
    }

    public void setPrestamo(Prestamo prestamo) {
        this.prestamo = prestamo;
    }

    public Kit getKit() {
        return kit;
    }

    public void setKit(Kit kit) {
        this.kit = kit;
    }

    public Componente getComponente() {
        return componente;
    }

    public void setComponente(Componente componente) {
        this.componente = componente;
    }

    public Integer getCantidad() {
        return cantidad;
    }

    public void setCantidad(Integer cantidad) {
        this.cantidad = cantidad;
    }

    public String getNumeroSerie() {
        return numeroSerie;
    }

    public void setNumeroSerie(String numeroSerie) {
        this.numeroSerie = numeroSerie;
    }
}

File: ti-lab-back/app/models/detalle_prestamo.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from app.models.base import Base


class DetallePrestamo(Base):
    __tablename__ = "detalles_prestamo"
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    prestamo_id = Column(Integer, ForeignKey("prestamos.id"), nullable=False)
    kit_id = Column(Integer, ForeignKey("kits.id"), nullable=True)
    componente_id = Column(Integer, ForeignKey("componentes.id"), nullable=True)
    cantidad = Column(Integer, nullable=False)
    numero_serie = Column(String(100), nullable=True)
    
    prestamo = relationship("Prestamo", back_populates="detalles")
    kit = relationship("Kit", back_populates="detalle_prestamos")
    componente = relationship("Componente", back_populates="detalle_prestamos")